src/kaleidoscope.c

```diff
@@ -34,12 +36,14 @@ int main(int argc, char **argv)
     
     // Setup optimizations.
     LLVMPassManagerRef pass_manager =  LLVMCreateFunctionPassManagerForModule(module);
-    LLVMAddTargetData(LLVMGetExecutionEngineTargetData(engine), pass_manager);
+    // LLVMAddTargetData(LLVMGetExecutionEngineTargetData(engine), pass_manager);
+#if 0
```

Comment:
May I suggest to check against the [version macros instead](https://github.com/llvm/llvm-project/blob/bbf2781bc49aee4d7ee8ec40dcf7316db360c454/llvm/include/llvm/Config/llvm-config.h.cmake#L144-L154)?
Perhaps something like:
```
#if (LLVM_VERSION_MAJOR < 18)
```

Reply:
I'll see if I can get that right, I'm totally new to LLVM (my interest in this project is as a way to learn it).  I think that the pass manager mechanisms changed a very long time ago (around LLVM 3.5 I seem to remember), so I'll have to hunt down when the deprecated APIs were removed.